### DISCUSSION TITLE
In which layer do socket/signalR and CronJob/Quartz should be in ?

### DISCUSSION THREAD

**Discussion by Nizar-Almshantaf** [jasontaylordev/CleanArchitecture#1119] [General]
I was wondering is there a version of this template with Socket/signalR and cronJob/quartz implementation? if not where should the implementation goes ? for example when coding a connectionHub in signalR in which layer should reside ?
Thanks,

**Comment by jasontaylordev** [comment]
For ASP.NET Core, I would implement SignalR within the Web project and have it interact with use cases (commands / queries) inside the Application layer.

**Reply by Nizar-Almshantaf** [reply]
thank you for your recommendation, I will implement it as suggested.

**Reply by Nizar-Almshantaf** [reply]
I updated the question, where should CronJobs/workers/Quartz reside? in which layer? I'm guessing Web, would you please assist  ? thanks

**Reply by jasontaylordev** [reply]
These processes, whether hosted in the Web project or in a new project, must depend on the Application project. For instance, you can schedule a Quartz.NET job to import data daily by calling the `BulkImportCommand` within the Application project. You can add more layers and projects to your solution, but ensure dependencies keep pointing inwards `BackgroundProcesses --> Application --> Domain`.
